### PR TITLE
Upgrade to last versions

### PR DESCRIPTION
**Pull request by rouazana**
- upgrade docker installation process (previous repository was no more available)
- fix security warnings and upgrade all plugins

### CODE REVIEW

**Review by blackheaven — APPROVED** [review]

**Review by remk — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

File: DockerfileSlave [hunk 1]
```diff
@@ -5,13 +5,13 @@ FROM jenkinsci/jnlp-slave:3.35-2-jdk11
 USER root
 
 RUN apt-get update \
-    && apt-get install -y apt-transport-https ca-certificates \
-    && apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D \
```
Comment (mbaechler) [inline DockerfileSlave:9]: we don't check the key anymore?

File: jenkins.yaml [hunk 2]
```diff
@@ -50,9 +50,9 @@ tool:
     installations:
       - name: "maven"
         properties:
-          - installSourceProperty:
+          - installSource:
               installers:
-              - mavenInstaller:
+              - maven:
                   id: 3.6.0
```
Comment (mbaechler) [inline jenkins.yaml:56]: 3.6.3?

File: DockerfileSlave (hunk 1 above)
Comment (rouazana) [reply DockerfileSlave:9]: the key is loaded on line 9 now:
```
curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
```

File: jenkins.yaml (hunk 2 above)
Comment (rouazana) [reply jenkins.yaml:56]: probably, but I don't want to restart the installation just to upgrade maven for now

File: DockerfileSlave (hunk 1 above)
Comment (mbaechler) [reply DockerfileSlave:9]: AFAIU, we fetched a known key before and now we just blindly accept the key the server gives us. It looks like a regression. Not sure it's a problem, but still, it deserves to ask if we can do like before or not

File: jenkins.yaml (hunk 2 above)
Comment (mbaechler) [reply jenkins.yaml:56]: I don't get the link between putting the last version in this file and restarting a server. Can't we check locally for example?

Comment (rouazana) [reply jenkins.yaml:56]: I don't see how. Testing this fully requires the jenkins server + 2 slaves, with one of them having the good btrfs configuration...
This PR reflects the last deployment, which has been tested. I propose to open an other PR to upgrade maven and maybe sbt

File: DockerfileSlave (hunk 1 above)
Comment (rouazana) [reply DockerfileSlave:9]: I just copy/paster the official documentation: https://docs.docker.com/engine/install/debian/ but that's right without checking that the key matches the expected one

Comment (rouazana) [reply DockerfileSlave:9]: this can be easily tested outside the process, but I suggest to fix this in an other issue

File: jenkins.yaml (hunk 2 above)
Comment (rouazana) [reply jenkins.yaml:56]: https://github.com/linagora/james-jenkins2/pull/11

File: DockerfileSlave (hunk 1 above)
Comment (rouazana) [reply DockerfileSlave:9]: and here it is: https://github.com/linagora/james-jenkins2/pull/12

**Review by chibenwa — APPROVED** [review]

**Comment by rouazana** [comment]
timeout, merging

**Pull request merged**